pkg/util/clientauthorizer/admin_test.go

```diff
@@ -0,0 +1,161 @@
+package clientauthorizer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+func TestAdminClientAuthorizer(t *testing.T) {
+	caBundlePath := "/fake/path/to/ca/cert.pem"
+	log := logrus.NewEntry(logrus.StandardLogger())
+
+	validCaKey, validCaCerts, err := utiltls.GenerateKeyAndCertificate("validca", nil, nil, true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	validIntermediateCaKey, validIntermediateCaCerts, err := utiltls.GenerateKeyAndCertificate("valid-intermediate-ca", validCaKey, validCaCerts[0], true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	invalidCaKey, invalidCaCerts, err := utiltls.GenerateKeyAndCertificate("invalidca", nil, nil, true, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, validSingleClientCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, validCertWithIntermediates, err := utiltls.GenerateKeyAndCertificate("validclient", validIntermediateCaKey, validIntermediateCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, invalidCommonNameClientCert, err := utiltls.GenerateKeyAndCertificate("invalidclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, invalidKeyUsagesCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	invalidKeyUsagesCert[0].ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth}
+
+	_, invalidExpiredCert, err := utiltls.GenerateKeyAndCertificate("validclient", validCaKey, validCaCerts[0], false, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	invalidExpiredCert[0].NotAfter = time.Now().Add(-time.Hour)
```

Comment:
I don't think you can really do this - I'd expect the certificate not to validate the signature it contains will now be invalid.  I think by here we're sort of testing Go library functionality rather than our functionality, so I think this case can be dropped.

Reply:
This works. We will get: "x509: certificate has expired or is not yet valid".

But I agree: it's not necessary because it is something that go standard library provides for us and should have test coverage for.